src/goose/brokers/sqs/broker.clj

```diff
@@ -0,0 +1,82 @@
+(ns goose.brokers.sqs.broker
+  (:require
+   [goose.broker :as b]
+   [goose.brokers.sqs.worker :as sqs-worker]
+   [goose.brokers.sqs.sqs-requests :as sqs-request]
+   [cognitect.aws.client.api :as aws]
+   [goose.utils :as u]))
+
+(defprotocol Close
+  "Closes connection to SQS Message Broker."
+  (close [this]))
+
+(defrecord SQS [client opts]
+  b/Broker
+  ;; SQS supports nothing more than:
+  ;;   - Enqueue a job
+  ;;   - Start a worker process
+  ;;   - Enqueued jobs API
+  ;;   - Batch jobs (Will implement later)
+
+  (enqueue
+    [this job]
+    (let [{:keys [client]} this
+          {:keys [queue-url]} opts]
+
+      (sqs-request/enqueue client queue-url (u/encode-to-str job))))
+
+  (start-worker
+    [this worker-opts]
+    (let [{:keys [client]} this
+          {:keys [queue-url]} opts]
+      (sqs-worker/start client queue-url worker-opts)))
+
+  Close
+  (close [_]
+    (println "SQS client does not require explicit closure.")))
```

Comment:
Close is not required to be implemented as part of Broker protocol if not needed.

Reply:
Okay!

Just a follow-up question - RMQ doesn't implements some methods from broker protocol. For instance the cron-job function. But still we can call it through `perform-every` of client. Though it ends up throwing a run-time error, Is there a way to let the developer know that they cannot do this(Just so their IDE/LSP can say them not to)(I know it is already mentioned in docs)